patches/0041-allow-lambdas-with-source_code_hash.patch

```diff
@@ -0,0 +1,141 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Venelin <venelin@pulumi.com>
+Date: Thu, 1 Feb 2024 20:07:48 +0000
+Subject: [PATCH 41/41] allow lambdas with source_code_hash
+
+
+diff --git a/internal/service/lambda/function.go b/internal/service/lambda/function.go
+index f7a47a0df7..2250da925a 100644
+--- a/internal/service/lambda/function.go
++++ b/internal/service/lambda/function.go
+@@ -169,7 +169,7 @@ func ResourceFunction() *schema.Resource {
+ 			"filename": {
+ 				Type:         schema.TypeString,
+ 				Optional:     true,
+-				ExactlyOneOf: []string{"filename", "image_uri", "s3_bucket"},
++				ExactlyOneOf: []string{"filename", "image_uri", "s3_bucket", "source_code_hash"},
+ 			},
+ 			"function_name": {
+ 				Type:         schema.TypeString,
+@@ -208,7 +208,7 @@ func ResourceFunction() *schema.Resource {
+ 			"image_uri": {
+ 				Type:         schema.TypeString,
+ 				Optional:     true,
+-				ExactlyOneOf: []string{"filename", "image_uri", "s3_bucket"},
++				ExactlyOneOf: []string{"filename", "image_uri", "s3_bucket", "source_code_hash"},
+ 			},
+ 			"invoke_arn": {
+ 				Type:     schema.TypeString,
+@@ -324,7 +324,7 @@ func ResourceFunction() *schema.Resource {
+ 			"s3_bucket": {
+ 				Type:         schema.TypeString,
+ 				Optional:     true,
+-				ExactlyOneOf: []string{"filename", "image_uri", "s3_bucket"},
++				ExactlyOneOf: []string{"filename", "image_uri", "s3_bucket", "source_code_hash"},
+ 				RequiredWith: []string{"s3_key"},
+ 			},
+ 			"s3_key": {
+@@ -491,7 +491,6 @@ func resourceFunctionCreate(ctx context.Context, d *schema.ResourceData, meta in
+ 		defer conns.GlobalMutexKV.Unlock(mutexKey)
+ 
+ 		zipFile, err := readFileContents(v.(string))
```

Comment:
What's happening here, what is v?

Reply:
https://github.com/hashicorp/terraform-provider-aws/blob/c42a6bbfd7f170bbb30254045c1657e2e9c973c2/internal/service/lambda/function.go#L946-L952

v is the `filename ` parameter here, I'm not patching this bit - it's just noise from my auto-formatter...